### DISCUSSION TITLE
What Typescript version is supported?

### DISCUSSION THREAD

**Discussion by arodik** [swc-project/swc#7814] [Q&A] [answered]
Does SWC work with any version of TypeScript? I looked through the documentation, but couldn't find any mentions of the required TypeScript version. It would be really helpful if there could be some information about compatible TypeScript versions on your website.

For instance, I'm currently using TypeScript 4.6.2 and I'm considering upgrading TS to version 5.1.6. However, I'm a bit uncertain whether SWC will accurately remove all type annotations. Could you provide more information on this, please?

**Comment by kwonoj** [comment]
Latest swc supposed to support latest stable typeacript, if not we keep it up asap.

**Comment by kdy1** [comment, marked as answer]
I documented it at https://swc.rs/docs/migrating-from-tsc#typescript-version
